internal/environment/checks.go

```diff
@@ -0,0 +1,36 @@
+// Copyright 2022 Juan Pablo Tosso and the OWASP Coraza contributors
+// SPDX-License-Identifier: Apache-2.0
+
+package environment
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/corazawaf/coraza/v3/debuglog"
+)
+
+// CheckFSAccess is a helper function to check if the WAF has access to the filesystem
+func CheckFSAccess(logger debuglog.Logger, tmpDir string) error {
+	if HasAccessToFS {
+		file, err := os.CreateTemp(tmpDir, "checkfsfile")
+		if err != nil {
+			return fmt.Errorf("create file: %w", err)
+		}
+		defer file.Close()
+
+		_, err = file.WriteString("check fs data")
```

Comment:
Not sure this check is 100% needed as CreateTemp is creating a file and opening it for writing hence not sure about the need of actually writing in it, also if we need that I'd write a single byte as that seems to be enough cc @anuraaga

Reply:
Thanks, I took a better look at CreateTemp, and your thoughts are reasonable. I've removed it for now